content/docs/refs-and-the-dom.md

```diff
@@ -51,49 +52,50 @@ class MyComponent extends React.Component {
 }
 ```
 
-### Accessing Refs {#accessing-refs}
+### Acessando Refs {#acessando-refs}
 
-When a ref is passed to an element in `render`, a reference to the node becomes accessible at the `current` attribute of the ref.
+Quando uma ref é passada para um elemento no `render`, uma referência para o nó se torna acessível no atributo `current` da ref.
 
 ```javascript
 const node = this.myRef.current;
 ```
 
-The value of the ref differs depending on the type of the node:
+O valor da ref difere dependendo do tipo do nó:
 
-- When the `ref` attribute is used on an HTML element, the `ref` created in the constructor with `React.createRef()` receives the underlying DOM element as its `current` property.
-- When the `ref` attribute is used on a custom class component, the `ref` object receives the mounted instance of the component as its `current`.
-- **You may not use the `ref` attribute on function components** because they don't have instances.
+- Quando o atributo `ref` é usado em um elemento HTML, a `ref` criada no construtor `React.createRef()` recebe um elemento DOM subjacente como a propriedade `current`.
 
-The examples below demonstrate the differences.
+- Quando o atributo `ref` é usado em um componente de classe, o objeto `ref` recebe uma instância montada de um componente
+  em sua propriedade `current`.
 
-#### Adding a Ref to a DOM Element {#adding-a-ref-to-a-dom-element}
+- **Você não pode usar o atributo `ref` em um componente funcional**, já que eles não possuem instâncias.
 
-This code uses a `ref` to store a reference to a DOM node:
+Os exemplos abaixo demonstram as diferenças.
+
+#### Adicionando uma Ref a um Elemento DOM {#adicionando-uma-ref-a-um-elemento-dom}
+
+Este código usa uma `ref` para armazenar uma referência a um nó DOM:
 
 ```javascript{5,12,22}
 class CustomTextInput extends React.Component {
   constructor(props) {
     super(props);
-    // create a ref to store the textInput DOM element
+    // Cria uma ref para armazenar o elemento textInput do DOM
     this.textInput = React.createRef();
     this.focusTextInput = this.focusTextInput.bind(this);
   }
 
   focusTextInput() {
-    // Explicitly focus the text input using the raw DOM API
-    // Note: we're accessing "current" to get the DOM node
+    // Explicitamente foca o input de texto usando a API do DOM diretamente
+    // Nota: nós estamos acessando o campo "current" para obter um nó do DOM.
     this.textInput.current.focus();
   }
 
   render() {
-    // tell React that we want to associate the <input> ref
-    // with the `textInput` that we created in the constructor
+    // Diz ao React que nós queremos associar o atributo ref do <input>
+    // com o `textInput` que nós criamos no construtor.
     return (
       <div>
-        <input
-          type="text"
-          ref={this.textInput} />
+        <input type="text" ref={this.textInput} />
```

Comment:
let's keep the code with the same lines as the original. this can be seem as silly but will break if the upstream updates this part